src/java.base/unix/native/libnio/ch/FileDispatcherImpl.c

```diff
@@ -375,7 +375,7 @@ Java_sun_nio_ch_FileDispatcherImpl_setDirect0(JNIEnv *env, jclass clazz,
         result = (int)file_stat.f_frsize;
     }
 #else
-    result == -1;
+    result = -1;
```

Comment:
This looks okay but can you add to the bug or the PR as to how you ran into this, is this a port to some platform that doesn't support it?

Reply:
I made a comment in the bug report.

Reply:
Thanks, I guessed there must be a new port in the picture as this case is not compiled in for any of the main stream platforms.